internal/adapters/operation/operation_storage_redis.go

```diff
@@ -365,3 +365,31 @@ func buildOperationFromMap(opMap map[string]string) (*operation.Operation, error
 		ExecutionHistory: executionHistory,
 	}, nil
 }
+
+func (r *redisOperationStorage) CleanExpiredOperations(ctx context.Context, schedulerName string) error {
+	operationsIDs, err := r.client.ZRangeByScore(ctx, r.buildSchedulerHistoryOperationsKey(schedulerName), &redis.ZRangeBy{
+		Min: "-inf",
+		Max: "+inf",
+	}).Result()
+
+	if len(operationsIDs) > 0 {
+
+		pipe := r.client.Pipeline()
+		for _, operationID := range operationsIDs {
+			operationExists := r.client.Exists(ctx, r.buildSchedulerHistoryOperationsKey(schedulerName)).Val()
```

Comment:
```suggestion
			operationExists := r.client.Exists(ctx, r.buildSchedulerOperationKey(schedulerName)).Val()
```
We are using the history key, but we should use `buildSchedulerOperationKey` that is the operation key